pkg/webhook/validate.go

```diff
@@ -166,22 +178,35 @@ func validatePolicyForNodeState(policy *sriovnetworkv1.SriovNetworkNodePolicy, s
 			}
 		}
 	}
+	return true, nil
+}
+
+func validatePolicyForNodePolicy(
+	current *sriovnetworkv1.SriovNetworkNodePolicy,
+	previous *sriovnetworkv1.SriovNetworkNodePolicy,
+) (bool, error) {
+	glog.V(2).Infof("validateConflictPolicy(): validate policy %s against policy %s",
+		current.GetName(), previous.GetName())
+
+	if current.GetName() == previous.GetName() {
+		return true, nil
+	}
 
-	for _, iface := range state.Spec.Interfaces {
-		if len(policy.Spec.NicSelector.PfNames) > 0 {
-			for _, pf := range policy.Spec.NicSelector.PfNames {
-				name, rngSt, rngEnd, err := sriovnetworkv1.ParsePFName(pf)
-				if err != nil {
-					return false, fmt.Errorf("invalid PF name: %s", pf)
-				}
-				if name == iface.Name {
-					for _, group := range iface.VfGroups {
-						if sriovnetworkv1.IndexInRange(rngSt, group.VfRange) || sriovnetworkv1.IndexInRange(rngEnd, group.VfRange) {
-							if group.PolicyName != policy.GetName() {
-								return false, fmt.Errorf("Vf index range in %s is overlapped with existing policies", pf)
-							}
-						}
-					}
+	for _, curPf := range current.Spec.NicSelector.PfNames {
+		curName, curRngSt, curRngEnd, err := sriovnetworkv1.ParsePFName(curPf)
+		if err != nil {
+			return false, fmt.Errorf("invalid PF name: %s", curPf)
+		}
+		for _, prePf := range previous.Spec.NicSelector.PfNames {
+			preName, preRngSt, preRngEnd, err := sriovnetworkv1.ParsePFName(prePf)
+			if err != nil {
```

Comment:
We don't need to handle the err for existing policies, right?

Reply:
It loops over all existing policies (exist in kube API) except the policy that has the same name as the policy being evaluated.

Reply:
I mean the err would not happen here. Since all these policies have been verified when it is created. Therefore we can just remove the error handling.

Reply:
Yes, you're right, removed the error checking for previous policy.